constants.bal

```diff
@@ -0,0 +1,254 @@
+# + BOUNDED - Reads might lag behind writes behind at most K updates of an item or by T time interval
+# + SESSION - Reads are guaranteed to honor the consistent-prefix, monotonic reads, monotonic writes, read-your-writes, 
+#             and write-follows-reads guarantees in a single client session.
+# + EVENTUAL - No ordering guarantee for reads. In the absence of any further writes, the replicas eventually converge.
+public enum ConsistencyLevel {
+    STRONG = "Strong",
+    BOUNDED = "Bounded",
+    SESSION = "Session",
+    EVENTUAL = "Eventual"
+}
+
+# Represents whether to include or exclude the document in indexing.
+#
+# + INCLUDE - Adds the document to the index
+# + EXCLUDE - Omits the document from indexing
+public enum IndexingDirective {
+    INCLUDE = "Include",
+    EXCLUDE = "Exclude"
+}
+
+# Represents the type of an Index.
+# 
+# + HASH - Useful for equality comparisons
+# + RANGE - Useful for equality, range comparisons and sorting
+# + SPATIAL - Useful for spatial queries
+public enum IndexType {
+    HASH = "Hash",
+    RANGE = "Range",
+    SPATIAL = "Spatial"
+}
+
+# The datatype for which the indexing behavior is applied to.
+# 
+# + STRING - Represents a string data type
+# + NUMBER - Represents a numeric data type
+# + POINT - Represents a point data type
+# + POLYGON - Represents a polygon data type
+# + LINESTRING - Represents a line string data type
+public enum IndexDataType {
+    STRING = "String",
+    NUMBER = "Number",
+    POINT = "Point",
+    POLYGON = "Polygon",
+    LINESTRING = "LineString"
+}
+
+# Type of operation that invokes the trigger.
+# 
+# + ALL - Trigger fires in all create, replace and delete operations
+# + CREATE - Trigger fires only in a create operations
+# + REPLACE - Trigger fires only in a replace operations
+# + DELETE - Trigger fires only in a delete operations
+public enum TriggerOperation {
+    ALL = "All",
+    CREATE = "Create",
+    REPLACE = "Replace",
+    DELETE = "Delete"
+}
+
+# When is the trigger is fired.
+# 
+# + PRE - Triggers fire before an operation 
+# + POST - Triggers fires after an operation
+public enum TriggerType {
+    PRE = "Pre",
+    POST = "Post"
+}
+
+# The access mode for the resource.
+# + ALL_PERMISSION - Provides read, write, and delete access to a resource
+# + READ_PERMISSION - Restricts the user to have only read access to the resource
+public enum PermisssionMode {
+    ALL_PERMISSION = "All",
+    READ_PERMISSION = "Read"
+}
+
+# The specific version for a given offer.
+# 
+# + PRE_DEFINED - Represents pre-defined throughput levels
+# + USER_DEFINED - Represents user-defined throughput levels
+public enum OfferVersion {
+    PRE_DEFINED = "V1",
+    USER_DEFINED = "V2"
+}
+
+# The performance levels for a specific throughput level.
+# 
+# + LEVEL_S1 - Performance level for pre-defined throughput level 
+# + LEVEL_S2 - Performance level for pre-defined throughput level
+# + LEVEL_S3 - Performance level for pre-defined throughput level
```

Comment:
It seems the same description is repeated for all 3 levels.

Reply:
Provided a detailed description for this